pkg/client/client.go

```diff
@@ -60,6 +61,10 @@ func (c *Client) Services(namespace string) ServiceInterface {
 	return newServices(c, namespace)
 }
 
+func (c *Client) Conditions(namespace string) ConditionsInterface {
```

Comment:
I was suggesting:

```
func ClientHasDesiredReplicas(client, controller) ConditionFunc {
  return func() bool, error {
  }
}
```

Instead.  That keeps the condition decoupled from the client interface and avoids the need for our fake objects to grow